### PR TITLE
status_server: reload TLS certificate on TLS errors (#12569)

### LINKED ISSUE #12546: HTTP status server does not automatically reload TLS certificate
## Bug Report

TLS server certificates for the HTTP status server are not automatically reloaded. The symptom is that, beyond the certificate validity period, Prometheus scrape requests will start failing with invalid certificate errors. Note that the certificates used by the gRPC server *are* properly reloaded; this bug only affects the status server.

It appears that the security configuration is only loaded once during server start: https://github.com/tikv/tikv/blob/b6220123ef333a9e7d7033a060d0c6adea6ad1b2/src/server/status_server/mod.rs#L505

There are no routines (either in `status_server/mod.rs` or in dependent libraries) to reload the server certificate for new connections.

### What version of TiKV are you using?

5.4.0

### What operating system and CPU are you using?

Linux x86_64, AWS EC2 i3.4xlarge

### Steps to reproduce

1. Create a certificate with a short validity period.
2. Start TiKV with TLS using this certificate.
3. Start Prometheus with a static HTTPS scrape target pointed to the TiKV metrics endpoint.
4. Wait until the certificate expires.
5. Replace the certificate file on disk with a newly issued, valid certificate, for both Prometheus and TiKV.
6. Notice that Prometheus scrapes now report errors for an invalid certificate due to an expired certificate.

### What did you expect?

I expect TiKV to have used the new certificate on disk for new connections, thus allowing Prometheus to continue to scrape the target without invalid certificate errors.

### What did happened?

TiKV's HTTP status server did not pick up the new certificate on disk, so it continued to terminate new TLS connections using the old certificate.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #12569 to release-5.4
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/12856
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/12856:release-5.4-470370510f3c
```

---

### What is changed and how it works?

Issue Number: Close #12546

What's Changed:

```commit-message
The commit implements auto TLS certificate reload for the status server.

The implementation is similar to ServerCredentialsFetcher of gRPC.
When there is a TLS error, we read the cert file metadata to check whether
the cert file has changed. If it changes, we re-initialize the SslContext
used for the HTTP server.
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Reload TLS certificate automatically when it changes.
```

### CODE REVIEW

File: src/server/status_server/mod.rs [hunk 1]
```diff
@@ -29,8 +29,15 @@ use hyper::server::Builder as HyperBuilder;
 use hyper::service::{make_service_fn, service_fn};
 use hyper::{self, header, Body, Method, Request, Response, Server, StatusCode};
 use online_config::OnlineConfig;
+<<<<<<< HEAD
```
Comment (Connor1996) [inline src/server/status_server/mod.rs:32]: Please fix the conflict @sticnarf

Comment (sticnarf) [reply src/server/status_server/mod.rs]: Done

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]

**Pull request merged**